### DISCUSSION TITLE
What does the "Enable pull diagnostics" experimental setting do?

### DISCUSSION THREAD

**Discussion by jnm2** [dotnet/roslyn#51649] [Q&A] [answered]
I'm just curious.

![image](https://user-images.githubusercontent.com/8040367/109892969-dce69f80-7c58-11eb-849d-78a9491fb597.png)

**Comment by jmarolf** [comment, marked as answer]
It has to do with the roslyn LSP implementation. With it checked VS will ask the LSP server for new data when it thinks it's appropriate instead of the LSP server pushing data to the client.
